### DISCUSSION TITLE
Logic for synced photo/video files in mobile apps?

### DISCUSSION THREAD

**Discussion by MichaelBui** [immich-app/immich#1541] [Q&A]
I'm asking this because I'm modifying the metadata of the images offline and I'm wondering how it will affect my Immich system:
1. I synced file A from Immich mobile app (iOS) to the server => file A marked as synced in the mobile app
2. Then I use the DAM app (DigiKam) to update the metadata (faces, labels, tags, rating,...) of the file A
3. Will it resync file A from the mobile app to the server again (and cause duplicated photos)?

What I guess:
- Immich use hashing (whole file) to identify unique file but once the metadata is updated, the hash will be changed => Immich will use the new hash and consider the existing file A from my mobile is a different file => it may sync ❌
- Immich mobile may have local storage to flag which file is uploaded/synced that will prevent the issue but if I delete the mobile app & install it again, the local storage will be deleted => it will sync ❌
- Immich may use DB to flag the file A in the device (by device ID) => it won't sync ✅ (unless device ID is also re-generated after the re-installation ❌)

Thanks in advance!

**Comment by alextran1502** [comment]
There is no mechanism to sync/detect change on the file level after the file is uploaded to Immich. So anything interaction that you perform on the file won't be recognized in Immich, and it can potentially create an issue with dedup-mechanism

**Reply by MichaelBui** [reply]
This I understand. My main concern is that after the metadata is modified, will the mobile app upload the file again?

**Reply by alextran1502** [reply]
No, the mobile app won't upload the files again.

**Reply by bo0tzz** [reply]
It might upload them again in some scenarios, like if you reinstall the app from another source (like f-droid). Currently the most likely behaviour is that a file won't be uploaded again, but that is not guaranteed, and could change in the future as we change the upload mechanism.

**Reply by MichaelBui** [reply]
Then, can we have some guideline on how to prevent re-uploading from mobile apps, especially since Immich are unable to manage metadata (faces, tags, labels, rating,...) at the moment & users are relying on other apps to do so?

**Reply by alextran1502** [reply]
There is no such guidelines for now as we are still moving things around and the mechanism of editing metadata will be likely added in the future.
